application/pages/modplatform/ftb/FtbModel.cpp

```diff
@@ -0,0 +1,259 @@
+#include <Json.h>
+#include "FtbModel.h"
+#include "MultiMC.h"
+
+#include "BuildConfig.h"
+#include "Env.h"
+
+namespace Ftb {
+
+ListModel::ListModel(QObject *parent) : QAbstractListModel(parent)
+{
+}
+
+ListModel::~ListModel()
+{
+}
+
+int ListModel::rowCount(const QModelIndex &parent) const
+{
+    return modpacks.size();
+}
+
+int ListModel::columnCount(const QModelIndex &parent) const
+{
+    return 1;
+}
+
+QVariant ListModel::data(const QModelIndex &index, int role) const
+{
+    int pos = index.row();
+    if(pos >= modpacks.size() || pos < 0 || !index.isValid())
+    {
+        return QString("INVALID INDEX %1").arg(pos);
+    }
+
+    ModpacksCH::Modpack pack = modpacks.at(pos);
+    if(role == Qt::DisplayRole)
+    {
+        return pack.name;
+    }
+    else if (role == Qt::ToolTipRole)
+    {
+        return pack.synopsis;
+    }
+    else if(role == Qt::DecorationRole)
+    {
+        if(m_logoMap.contains(pack.name))
+        {
+            return (m_logoMap.value(pack.name));
+        }
+
+        for(auto art : pack.art) {
+            if(art.type == "square") {
+                ((ListModel *)this)->requestLogo(pack.name, art.url);
+            }
+        }
+
+        QIcon icon = MMC->getThemedIcon("screenshot-placeholder");
+        return icon;
+    }
+    else if(role == Qt::UserRole)
+    {
+        QVariant v;
+        v.setValue(pack);
+        return v;
+    }
+
+    return QVariant();
+}
+
+void ListModel::performPaginatedSearch()
```

Comment:
Should rename this, as the search isn't paginated.

To add onto this, the API merely returns a list of pack IDs (that need to then be each queried) - we can implement pagination by batching this process, querying for X amount at a time (from the list of IDs we currently have).

If I get chance, I'll look into how pagination with the ListModel works, else this may form a future pr.